src/core/storage/sframe_interface/unity_sframe.cpp

```diff
@@ -1200,6 +1201,73 @@ unity_sframe::random_split(float percent, uint64_t random_seed, bool exact) {
   return logical_filter_split(logical_filter_array);
 }
 
+std::shared_ptr<unity_sframe_base> unity_sframe::shuffle() {
+#include <iostream>  // TODO: remove
+  
+  log_func_entry();
+
+  std::shared_ptr<unity_sframe> temp(new unity_sframe());
+
+  size_t length = this->size();
+
+  std::vector<std::string> column_names = this->column_names();
+
+  const size_t num_buckets = 2;   // XXX: how to determine the number of buckets?
```

Comment:
This is hard to do perfect.  I'd create a global variable, call it TURI_SFRAME_SHUFFLE_BUCKET_SIZE and then just do  I'd just do num_rows() / TURI_SFRAME_SHUFFLE_BUCKET_SIZE.  Default to like 512.

Reply:
Good idea. Done.